File: Algorithms/128.LongestConsecutiveSequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        m = set(nums)
        longestLength = 0
        for i in nums:
            length = 1
            j = i+1
            while j in m:
                m.remove(j)
                j += 1
                length += 1
            j = i-1
            while j in m:
                m.remove(j)
                j -= 1
                length += 1
            longestLength = max(longestLength, length)
        return longestLength

File: Algorithms/128.LongestConsecutiveSequence.java
public class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> m = new HashSet<>();
        for (int i: nums)
            m.add(i);
        int longestLength = 0;
        for (int i: nums) {
            int length = 1;
            for (int j=i+1; m.contains(j); j++) {
                m.remove(j);
                length ++;
            }
            for (int j=i-1; m.contains(j); j--) {
                m.remove(j);
                length ++;
            }
            longestLength = Math.max(longestLength, length);
        }
        return longestLength;
    }
}
